packages/runner/lib/sdk/locks.ts

```diff
@@ -2,62 +2,336 @@
 import type { Result } from '@nangohq/utils';
 import { Err, Ok } from '@nangohq/utils';
 
-interface Lock {
-    key: string;
-    owner: string;
-    expiresAt: Date;
-}
+const MAX_LOCKS = 256;
```

Comment:
this implementation only allows for 256 different locks. Given how niche of a feature it is I think it is a acceptable limitation